modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/manager/CETManagerImpl.java

```diff
@@ -95,7 +95,17 @@ public CET getCET(long companyId, String externalReferenceCode) {
 
 		Map<String, CET> cetsMap = _getCETsMap(companyId);
 
-		return cetsMap.get(externalReferenceCode);
+		CET cet = cetsMap.get(externalReferenceCode);
+
+		if (cet == null) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					"No CET found for external reference code " +
+						externalReferenceCode);
+			}
```

Comment:
In case `Log4jLogImpl` is used as instance for Log, it seems the level is already checked internally:
```
@Override
	public void warn(Object msg) {
		_logger.logIfEnabled(_logWrapperClassName, Level.WARN, null, msg, null);
	}
```

Reply:
Yeah, but we are required to add the `if` (usually by SF) because the computation to create the log message may take long, so an `if` is more performant. I know it's picky, but on the other hand, there's no harm in adding the `if`, I guess.

Reply:
oh cool, makes sense then